WordPress/WordPressTest/TestUtilities/JSONObject.swift

```diff
@@ -7,23 +7,34 @@ extension JSONObject {
 
     /// Loads the specified json file and returns a dictionary representing it.
     ///
-    /// - Parameter fileName: The full name of the json file to load.
+    /// - Parameter fileName: The name of the json file to load. The "json" file extension can be omitted.
     /// - Returns: A dictionary representing the contents of the json file.
-    static func loadFile(named fileName: String) throws -> JSONObject {
-        return try loadFile(
-            (fileName as NSString).deletingPathExtension,
-            type: (fileName as NSString).pathExtension
-        )
-    }
+    static func loadJSONFile(named fileName: String) throws -> JSONObject {
+        let type = (fileName as NSString).pathExtension
+        if type != "" && type != "json" {
+            throw NSError(
+                domain: "JSONObject",
+                code: 1,
+                userInfo: [
+                    NSLocalizedDescriptionKey: "File extension should be 'json': \(fileName)"
+                ]
+            )
+        }
 
-    /// Loads the specified JSON file and returns a dictionary representing it.
-    ///
-    /// - Parameters:
-    ///   - name: The name of the file
-    ///   - type: The extension of the file
-    /// - Returns: A dictionary representing the contents of the JSON file.
-    static func loadFile(_ name: String, type: String) throws -> JSONObject {
-        let url = try XCTUnwrap(Bundle(for: BundlerFinder.self).url(forResource: name, withExtension: type))
+        let testBundle = Bundle(for: BundlerFinder.self)
+        let candidates = [
+            testBundle.url(forResource: fileName, withExtension: nil),
+            testBundle.url(forResource: fileName, withExtension: "json"),
+        ]
+        guard let url = candidates.compactMap({ $0 }).first else {
+            throw NSError(
+                domain: "JSONObject",
+                code: 1,
+                userInfo: [
+                    NSLocalizedDescriptionKey: "Can't find JSON file named \(fileName) or \(fileName).json"
+                ]
+            )
```

Comment:
Same question about reaching for `NSError`, though a programmer could enter a valid-looking path that nonetheless fails. 

I wonder if this safety check and the one above could be combined?

Reply:
The file extension was explicitly checked based on https://github.com/wordpress-mobile/WordPress-iOS/pull/18507#discussion_r865605510, which I agree to, since if a file name that dose exist but doesn't end with "json" is passed to `loadJSONFile` function, it's very likely the file name should be corrected.
> if input "file-name.extension", verify extension is JSON and throw an error otherwise.